definitions/features/foreman_proxy.rb

```diff
@@ -3,19 +3,62 @@ class Features::ForemanProxy < ForemanMaintain::Feature
     label :foreman_proxy
   end
 
-  attr_reader :dhcpd_conf_file, :cert_path
+  attr_reader :dhcpd_conf_file, :cert_path, :request_url,
+              :default_ssl_cert_options
 
   def initialize
     @dhcpd_conf_file = '/etc/dhcp/dhcpd.conf'
     @cert_path = ForemanMaintain.config.foreman_proxy_cert_path
+    @request_url = "https://#{hostname}:9090/dhcp"
+    @default_ssl_cert_options = load_default_ssl_cert_options
   end
 
   def valid_dhcp_configs?
     dhcp_req_pass? && !syntax_error_exists?
   end
 
+  def list_reservations(subnet)
+    resp = dhcp_restclient["/#{subnet}"].get { |response, _request, _result| response }
```

Comment:
Another option I think could be `ForemanMaintain::Utils::DhcpReservation` with instance method `#list #add #delete`